src/components/MoneyReportHeader.tsx

```diff
@@ -373,21 +317,226 @@ function MoneyReportHeader({policy, report: moneyRequestReport, transactionThrea
         Navigation.navigate(ROUTES.PROCESS_MONEY_REQUEST_HOLD.getRoute(Navigation.getReportRHPActiveRoute()));
     }, [dismissedHoldUseExplanation, isLoadingHoldUseExplained, isOnHold]);
 
-    useEffect(() => {
-        if (canDeleteRequest) {
-            return;
+    const primaryAction = useMemo(() => {
+        if (!moneyRequestReport) {
+            return '';
+        }
+        return getReportPrimaryAction(moneyRequestReport, transactions, violations, policy);
+    }, [moneyRequestReport, policy, transactions, violations]);
+
+    const primaryActions = {
+        [CONST.REPORT.PRIMARY_ACTIONS.SUBMIT]: (
+            <Button
+                success
+                text={translate('common.submit')}
+                onPress={() => {
+                    if (!moneyRequestReport) {
+                        return;
+                    }
+                    submitReport(moneyRequestReport);
+                }}
+            />
+        ),
+        [CONST.REPORT.PRIMARY_ACTIONS.APPROVE]: (
+            <Button
+                success
```

Comment:
We don't have any onPress handler for this action?

Reply:
Handler is defined one line below 😅

Reply:
Ohhh sorry I don't know how I missed it 😓

Reply:
It happens to the best ;)